polaris/hub/client.py

```diff
@@ -623,6 +632,85 @@ def _upload_dataset(
 
             return response
 
+    def _upload_v2_dataset(
+        self,
+        dataset: DatasetV2,
+        access: AccessType = "private",
+        timeout: TimeoutTypes = (10, 200),
+        owner: HubOwner | str | None = None,
+        if_exists: ZarrConflictResolution = "replace",
+    ):
+        """
+        Upload a V2 dataset to the Polaris Hub.
+        """
+
+        with ProgressIndicator(
+            start_msg="Uploading artifact...",
+            success_msg="Uploaded artifact.",
+            error_msg="Failed to upload dataset.",
+        ) as progress_indicator:
+            # Check if a dataset license was specified prior to upload
+            if not dataset.license:
+                raise InvalidDatasetError(
+                    f"\nPlease specify a supported license for this dataset prior to uploading to the Polaris Hub.\nOnly some licenses are supported - {get_args(SupportedLicenseType)}."
+                )
+
+            # Normalize timeout
+            if timeout is None:
+                timeout = self.settings.default_timeout
```

Comment:
We should move the license and timeout checks (lines 653 and 659, respectively) to the calling `upload_dataset` function. They're shared right now and I doubt the logic will change.